app/service/scorecard-service.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+
+const angular = require('angular');
+
+angular.module('appShooter').factory('scorecardService', ['$log', '$q','$window', '$http', scorecardService]);
+
+
+function scorecardService($log, $q, $window, $http) {
+  $log.debug('entered scorecardService');
+  let service = {};
+  let token = $window.localStorage.getItem('token');
+  service.data = [];
+  service.competitions = [];
+
+
+  service.createCompetition = function(data) {
+    $log.debug('entered service.createCompetition');
+    let url =`${__API_URL__}/api/competition`;
+
+    let config = {
+      headers: {
+        'Content-Type':'application/json',
+        'Authorization':`Bearer ${token}`
+      }
+    };
+    return $http.post(url, data, config)
+    .then(res => {
+      $log.info('Success', res.data);
+      return (res.data);
+    })
+    .catch(err => {
+      $log.error('Failed to return createCompetition data', err);
+      return $q.reject(err);
+    });
+  };
+
+
+  service.createMatches = function(matchModel, competitionId) {
+    $log.debug('entered service.createMatches');
+
+    let url =`${__API_URL__}/api/competition/${competitionId}/match`;
+
+    let config = {
+      headers: {
+        'Content-Type':'application/json',
+        'Authorization':`Bearer ${token}`
+      }
+    };
+
+
+    let matches = [];
+    for (var i = 1; i < 4; i++ ){
+      var newMatch = angular.copy(matchModel);
+      newMatch.matchNumber = i;
+      matches.push($http.post(url, newMatch, config));
+    }
+
+    return $q.all(matches)
+    .then((matches) => {
+      return $q.resolve(matches);
+    })
+    .catch( err => {
+      return $q.reject(err);
+    });
+  };
+
+  service.createMatchShots = function(competitionId, matches, allMatchScores, shotModel) {
+    $log.debug('entered service.createMatchShots');
+
+    var config = {
+      headers: {
+        'Content-Type':'application/json',
+        'Authorization':`Bearer ${token}`
+      }
+    };
+
+    var shots = [];
+
+    for (var i = 0; i < 3; i++ ){
```

Comment:
THis is awesome, But it could have been easier to refactor/ or add a new route on the backend that would take a whole competition rather that use this" upload as you go" route style